scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -697,9 +696,11 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
 
         val rawtss = new ListBuffer[List[Tree]]
         while (!hasTypes && !hasImplicits && token.is[LeftParen]) {
-          if (openParenPos != in.tokenPos && secondOpenParenPos == 0)
-            secondOpenParenPos = in.tokenPos
-          accept[LeftParen]
+          if (rawtss.nonEmpty) {
```

Comment:
Do we need to do this in a while then? Since in every step we add to `rawtss` the loop will effectively only run once, no?

Should we replace it with an `if` ?

Reply:
@tgodzik done. @dos65 added tests.